Add migration test 3.3 -> 3.4

burqen opened the pull request.

Commit:
Add migration test 3.3 -> 3.4

Commit:
Also verify additional updates to old indexes

tinwelint commented on community/neo4j/src/test/java/migration/StartOldDbOn3_4AndCreateFusionIndexIT.java:
```diff
@@ -155,6 +239,17 @@ private void verifyIndexes( GraphDatabaseAPI db, Label label ) throws Exception
 
+    private void verifyAfterAdditionalUpdate( GraphDatabaseAPI db, Label label ) throws Exception
+    {
+        // There should be an index for the label and KEY1 containing 100 nodes
```
Comment is wrong, I realize that it's twice those numbers now that one more round of updates have been added. Comment could reflect that, or be removed.

tinwelint commented on community/neo4j/src/test/java/migration/StartOldDbOn3_4AndCreateFusionIndexIT.java:
```diff
@@ -155,6 +239,17 @@ private void verifyIndexes( GraphDatabaseAPI db, Label label ) throws Exception
+        assertTrue( hasIndex( db, label, KEY1 ) );
+        assertEquals( 200, countIndexedNodes( db, label, KEY1 ) );
+
+        // There should be an index for the label and KEY1+KEY2 containing 34 nodes
```
Likewise here

tinwelint reviewed (commented):
Should the test also assert that the old index is still for the old index provider?

Commit:
Verify IndexProvider in index migration test

Had to rebuild 3.3 index to make labels used when creating the 3.2 store
align with expected provider.

burqen commented:
@tinwelint Comments addressed and rebase complete
